src/main/java/uk/gov/companieshouse/service/CompanyService.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.companieshouse.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.util.UriTemplate;
+import uk.gov.companieshouse.api.ApiClient;
+import uk.gov.companieshouse.api.ApiClientService;
+import uk.gov.companieshouse.api.error.ApiErrorResponseException;
+import uk.gov.companieshouse.api.handler.exception.URIValidationException;
+import uk.gov.companieshouse.api.model.ApiResponse;
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.exception.ServiceException;
+import uk.gov.companieshouse.mapper.CompanyClosableMapper;
+
+@Service
+public class CompanyService {
+    private static final UriTemplate GET_COMPANY_URI = new UriTemplate("/company/{companyNumber}");
+
+    private final ApiClientService apiClientService;
+
+    private final CompanyClosableMapper companyClosableMapper;
+
+    @Autowired
+    public CompanyService(CompanyClosableMapper companyClosableMapper, ApiClientService apiClientService) {
+        this.companyClosableMapper = companyClosableMapper;
+        this.apiClientService = apiClientService;
+    }
+
+    public CompanyProfileApi getCompanyProfile(String companyNumber) throws ServiceException {
```

Comment:
same here, feels like this should be moved into a separate CompanyProfileClient class, similarly to how the frontend is structured.

Reply:
so should this method be hollowed out and pass the request to the client class or should I call the client directly In the controller?

Reply:
Hmmmm, I think it's fine to call the client class directly tbh.

The reason I introduced the DissolutionService facade layer was due to the fact we would regularly be extending it, although I don't see us extending the Company Client much beyond this.